code/modules/library/book.dm

```diff
@@ -0,0 +1,370 @@
+				return 1
+			else
+				author = newauthor
+		if("Summary")
+			var/newsummary = strip_html(input(usr, "Write the new summary:") as message|null, MAX_SUMMARY_LEN)
+			if(!newsummary)
+				to_chat(usr, "The name is invalid.")
+				return
+			else
+				summary = newsummary
+		if("Edit Current Page")
+			if(carved)
+				to_chat(usr, "<span class='notice'>The pages of [title] have been cut out!</span>")
+				return
+			if(!current_page)
+				to_chat(usr, "<span class='notice'>You need to turn to a page before writing in the book.</span>")
+				return
+			var/character_space_remaining = MAX_CHARACTERS_PER_BOOKPAGE - length(pages[current_page])
+			if(character_space_remaining <= 0)
+				to_chat(usr, "<span class='notice'>There's not enough space left on this page to write anything!</span>")
+				return
+			var/content = strip_html(input(usr, "Add Text to this page, you have [character_space_remaining] characters of space left:") as message|null, MAX_CHARACTERS_PER_BOOKPAGE)
+			if(!content)
+				to_chat(usr, "The content is invalid.")
+				return
+			//check if length of current text content + what player is adding is larger than our character limit
+			else if((length(content) + length(pages[current_page])) > MAX_CHARACTERS_PER_BOOKPAGE)
+				//if true, let's cut down the text to fit perfectly into our character limit, player is only half-pissed!
+				pages[current_page] += dd_limittext(content, (MAX_CHARACTERS_PER_BOOKPAGE - length(pages[current_page])))
+			else
+				pages[current_page] += content
+		if("Add Page")
+			if(carved)
+				to_chat(usr, "<span class='notice'>You can't add anymore pages, the pages of [title] have been cut out and the book is ruined!</span>")
+				return
+			if(length(pages) >= MAX_PAGES)
+				to_chat(usr, "<span class='notice'>You can't fit anymore pages in this book!</span>")
+				return
+			to_chat(usr, "<span class='notice'>You add another page to the book!</span>")
+			pages += " "
+		if("Remove Page")
+			if(!length(pages))
+				to_chat(usr, "<span class='notice'>There aren't any pages in this book!</span>")
+				return
+			var/page_choice = stripped_input(usr, "There are [length(pages)] pages, which page number would you like to remove?")
+			if(!isnum(page_choice) || page_choice > length(pages))
+				to_chat(usr, "That is not an acceptable value.")
+				return
+			pages -= pages[page_choice]
+
+/obj/item/book/proc/edit_page(content, page_number)
+	if(!content)
+		return
+	if(length(pages) < page_number) //can't remove the page if it doesn't exist
+		return
+	pages += content
+
+/obj/item/book/proc/add_page(content)
+	var/text = content
+	if(!text)
+		text = " "
+	if(length(pages) >= MAX_PAGES)
+		return FALSE
+	pages += text
+	current_page = length(pages) //open to newely added page so player can edit it
+	return TRUE
+
+/obj/item/book/proc/remove_page(page_number)
+	if(length(pages) < page_number) //can't remove the page if it doesn't exist
+		return FALSE
+	if(pages -= pages[page_number])
```

Comment:
The check here seems fishy. You're removing something from the list and then checking the list itself?

Reply:
I'm just as confused as you are, this is more 2AM code™

Reply:
cleaned up